bundles/net.tourbook.device/src/net/tourbook/device/sporttracks/FitLogSAXHandler.java

```diff
@@ -837,7 +834,7 @@ private void finalizeTour_10_SetTourType(final TourData tourData) {
 
    private void finalizeTour_20_SetTags(final TourData tourData) {
 
-      final ArrayList<Equipment> equipmentNames = _currentActivity.equipmentNames;
+      final ArrayList<Equipment> equipmentNames = _currentActivity._equipmentNames;
       if (equipmentNames.size() == 0) {
          return;
       }
```

Comment:
Why is this code modified?

Especially`!_equipments.isEmpty()` is a double negation (not no content) which is not very easy to read

![mt-double-negation](https://user-images.githubusercontent.com/1283445/103751703-ff698d00-5008-11eb-8802-f1705ed094f2.png)

Reply:
because @FJBDev  recommended to use "isEmpty" instead of "size()==0" for better performance.

Reply:
When you are testing` size() ==0`, then you should use `isEmpty().` But per Wolfgang's request [here](https://github.com/wolfgang-ch/mytourbook/pull/257#issuecomment-738204815), we should not use double negation as there is no way at the moment to say `.hasContent()` like for [StringUtils](https://github.com/wolfgang-ch/mytourbook/blob/30a22d0ed54aaedb61537c3816c0a5e09978e32f/bundles/net.tourbook.common/src/net/tourbook/common/util/StringUtils.java#L29)

Reply:
ok, I'll revert back the modification